packages/runtime-common/indexer.ts

```diff
@@ -503,6 +529,126 @@ export class Indexer {
     ];
   }
 
+  async searchPrerendered(
+    realmURL: URL,
+    { filter, sort, page }: Query,
+    loader: Loader,
+    opts: QueryOptions = {},
+  ): Promise<{
+    prerenderedCards: PrerenderedCard[];
+    prerenderedCardCssItems: PrerenderedCardCssItem[];
+    meta: QueryResultsMeta;
+  }> {
+    let { results, meta } = await this._search(
+      realmURL,
+      { filter, sort, page },
+      loader,
+      opts,
+      'SELECT url, ANY_VALUE(file_alias) as file_alias, ANY_VALUE(embedded_html) as embedded_html',
+    );
+
+    let cardFileAliases = results.map((c) => c.file_alias);
+    let realmVersion = meta.page.realmVersion;
+
+    let conditions: CardExpression[] = [
+      ['i.realm_version = ', param(realmVersion)],
+      ['i.type =', param('instance')],
+      [
+        `i.file_alias IN (${cardFileAliases
+          .map((fileAlias) => `'${fileAlias}'`)
+          .join(', ')})`,
+      ],
+    ];
+
+    // This query will give us a list of indexed css entries and their corresponding instances that have that particular css module as a dependency
+    let cssSourceQuery = [
+      `WITH instance_deps AS (
+        SELECT
+          file_alias AS instance_file_alias,
+          jsonb_array_unnested AS dep_url_file_alias -- 'jsonb_array_unnested' is named like that on purpose so that our sqlite adjuster knows to transform it to json_each.value to keep compatibility with sqlite
+        FROM boxel_index i,
+          jsonb_array_elements_text(deps) AS jsonb_array_unnested
+        WHERE`,
+      ...every(conditions),
+      `),
+      css_deps AS (
+        SELECT
+          bi.source AS css_source,
+          bi.file_alias AS css_module_id,
+          array_agg(id.instance_file_alias) AS instances
+        FROM boxel_index bi
+        JOIN instance_deps id ON bi.file_alias = id.dep_url_file_alias
+        WHERE bi.type = 'css'
+        GROUP BY bi.source, bi.file_alias
+      )
+      SELECT
+        css_module_id,
+        array_to_json(instances) AS instances,
+        css_source
+      FROM css_deps;
+      `,
+    ];
+
+    let groupedCssInstanceData = (await this.queryCards(
```

Comment:
This query fetches unique css and tells which instances use it.  Example output of this query:

<img width="1168" alt="image" src="https://github.com/cardstack/boxel/assets/273660/951cf1c4-2b49-4561-9bb8-74ec1be3f183">